sources/packages/backend/apps/api/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -103,4 +103,49 @@ export class CASSupplierService {
 
     return student.casSupplier;
   }
+
+  /**
+   * Retries CAS supplier info for a student.
+   * Inserts a new CAS Supplier record for the student with PendingSupplierVerification status.
+   * @param studentId student id.
+   * @param auditUserId user id for the record creation.
+   */
+  async retryCASSupplier(
+    studentId: number,
+    auditUserId: number,
+  ): Promise<CASSupplier> {
+    const student = await this.studentService.getStudentById(studentId);
+    const auditUser = { id: auditUserId } as User;
+    if (!student) {
+      throw new CustomNamedError("Student not found.", STUDENT_NOT_FOUND);
+    }
```

Comment:
This should be handled at the controller lever and converted to a proper HTTP error.